src/components/documentManager.ts

```diff
@@ -0,0 +1,611 @@
+
+        if (item != null) {
+            if (item.BackdropImageTags && item.BackdropImageTags.length) {
+                // get first backdrop of image if applicable
+                src = JellyfinApi.createUrl(
+                    `Items/${item.Id}/Images/Backdrop/0?tag=${item.BackdropImageTags[0]}`
+                );
+            } else if (
+                item.ParentBackdropItemId &&
+                item.ParentBackdropImageTags &&
+                item.ParentBackdropImageTags.length
+            ) {
+                // otherwise get first backdrop from parent
+                src = JellyfinApi.createUrl(
+                    `Items/${item.ParentBackdropItemId}/Images/Backdrop/0?tag=${item.ParentBackdropImageTags[0]}`
+                );
+            }
+        }
+
+        const element: HTMLElement | null = document.querySelector(
+            '#waiting-container-backdrop'
+        );
+
+        if (element === null) {
+            console.error(
+                'documentManager: Cannot find #waiting-container-backdrop'
+            );
+        } else {
+            (<HTMLElement>element).style.backgroundImage =
+                src != null ? `url(${src})` : '';
+        }
+    }
+
+    /**
+     * Set a random backdrop on the waiting container
+     *
+     * @returns {Promise<void>} promise waiting for the backdrop to be set
+     */
+    private static setRandomUserBackdrop(): Promise<void> {
+        return JellyfinApi.authAjaxUser('Items', {
+            dataType: 'json',
+            type: 'GET',
+            query: {
+                SortBy: 'Random',
+                IncludeItemTypes: 'Movie,Series',
+                ImageTypes: 'Backdrop',
+                Recursive: true,
+                Limit: 1,
+                // Although we're limiting to what the user has access to,
+                // not everyone will want to see adult backdrops rotating on their TV.
+                MaxOfficialRating: 'PG-13'
+            }
+        }).then((result) => {
+            if (result.Items && result.Items[0])
+                return DocumentManager.setWaitingBackdrop(result.Items[0]);
+            else return DocumentManager.setWaitingBackdrop(null);
+        });
+    }
+
+    /**
+     * Stop the backdrop rotation
+     */
+    public static clearBackdropInterval(): void {
+        if (this.backdropTimer !== 0) {
+            clearInterval(this.backdropTimer);
+            this.backdropTimer = 0;
+        }
+    }
+
+    /**
+     * Start the backdrop rotation, restart if running, stop if disabled
+     *
+     * @returns {Promise<void>} promise for the first backdrop to be set
+     */
+    public static startBackdropInterval(): Promise<void> {
+        // avoid running it multiple times
+        this.clearBackdropInterval();
+
+        // skip out if it's disabled
+        if (!this.backdropPeriodMs) {
+            this.setWaitingBackdrop(null);
+            return Promise.resolve();
+        }
+
+        this.backdropTimer = <any>(
```

Comment:
This makes the linter unhappy, but after searching about the setInterval function, I came to the conclusion that it's a mess and I would need help to fix this

Reply:
I think we can merge despite this. It's just the only linter warning I couldn't remove.

Reply:
This is a number, TypeScript is not wrong about the return type: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval#return_value

Reply:
Typescript thinks this is not a number

Reply:
```
ERROR in /home/hawken/git/mirror/jellyfin-chromecast/src/components/documentManager.ts
./components/documentManager.ts
[tsl] ERROR in /home/hawken/git/mirror/jellyfin-chromecast/src/components/documentManager.ts(352,9)
      TS2322: Type 'Timeout' is not assignable to type 'number'.
 @ ./components/maincontroller.ts 8:0-52 91:8-45
 @ ./app.ts 2:0-37
```

Reply:
https://stackoverflow.com/questions/45802988/typescript-use-correct-version-of-settimeout-node-vs-window

Reply:
This issue is beyond me, I really don't want to do some other hack instead of this hack. If we are going for a hack I'll add a comment here to tell the linter to leave it alone.